src/AppContainer.dev.js

```diff
@@ -49,7 +49,7 @@ class AppContainer extends Component {
     // Hot reload has finished.
     // Force-update the whole tree, including
     // components that refuse to update.
-    if (prevProps.component !== this.props.component) {
+    if (prevProps.children !== this.props.children) {
```

Comment:
I think we want to compare `props.children.type` and force-update if it changed. Otherwise, if the user re-renders from the top for other reasons than a hot reload, we are wasting force updates. Sure, the user probably wouldn’t do that, but it’s better to double check.